lib/drivers/medtronic600/NGPHistoryParser.js

```diff
@@ -1566,29 +1562,34 @@ class NGPHistoryParser {
     if (lastTempBasalEvent &&
       lastTempBasalEvent.eventType === NGPHistoryEvent.EVENT_TYPE.TEMP_BASAL_PROGRAMMED) {
       const suppressedBasal = this.findSuppressedBasal(lastTempBasalEvent);
-      const basal = this.cfg.builder.makeTempBasal()
-        .with_rate(lastTempBasalEvent.rate)
-        .with_expectedDuration(lastTempBasalEvent.expectedDuration)
-        .with_duration(lastTempBasalEvent.expectedDuration)
-        .set('suppressed', {
-          type: 'basal',
-          deliveryType: 'scheduled',
-          rate: suppressedBasal.rate,
-          scheduleName: suppressedBasal.patternName,
-        });
+      // Don't create the temp basal if we don't find the underlying suppressed info
+      // This could happen if the ONLY basal event we retrieve from the pump history
+      // is a TEMP_BASAL_PROGRAMMED.
+      if (!_.isUndefined(suppressedBasal)) {
+        const basal = this.cfg.builder.makeTempBasal()
+          .with_rate(lastTempBasalEvent.rate)
+          .with_expectedDuration(lastTempBasalEvent.expectedDuration)
+          .with_duration(lastTempBasalEvent.expectedDuration)
+          .set('suppressed', {
+            type: 'basal',
+            deliveryType: 'scheduled',
+            rate: suppressedBasal.rate,
+            scheduleName: suppressedBasal.patternName,
+          });
 
-      // We need to find the normal rate here to calculate event.rate, because
-      // if it's a PERCENT record, event.rate will be 0.
-      if (lastTempBasalEvent.type === NGPUtil.NGPConstants.TEMP_BASAL_TYPE.PERCENT) {
-        basal
-          .with_percent(lastTempBasalEvent.percentageOfRate / 100.0)
-          .with_rate(suppressedBasal.rate * (lastTempBasalEvent.percentageOfRate / 100.0));
-      }
+        // We need to find the normal rate here to calculate event.rate, because
+        // if it's a PERCENT record, event.rate will be 0.
+        if (lastTempBasalEvent.type === NGPUtil.NGPConstants.TEMP_BASAL_TYPE.PERCENT) {
+          basal
+            .with_percent(lastTempBasalEvent.percentageOfRate / 100.0)
+            .with_rate(suppressedBasal.rate * (lastTempBasalEvent.percentageOfRate / 100.0));
```

Comment:
If you do decide to follow previous suggestion, you'll have to use the current pump settings to determine the rate in the case where suppressed info is not available.

Reply:
See above comment